src/panels/calendar/ha-panel-calendar.ts

```diff
@@ -0,0 +1,281 @@
+import { getCalendars, fetchCalendarEvents } from "../../data/calendar";
+
+@customElement("ha-panel-calendar")
+class PanelCalendar extends LitElement {
+  @property() public hass!: HomeAssistant;
+
+  @property({ type: Boolean, reflect: true })
+  public narrow!: boolean;
+
+  @property() private _calendars: SelectedCalendar[] = [];
+
+  @property() private _events: CalendarEvent[] = [];
+
+  private _start?: Date;
+
+  private _end?: Date;
+
+  protected firstUpdated(changedProps) {
+    super.firstUpdated(changedProps);
+    if (!this.hass) {
+      return;
+    }
+
+    this._calendars = getCalendars(this.hass).map((calendar) => ({
+      selected: true,
+      calendar,
+    }));
+
+    if (!this._start || !this._end) {
+      return;
+    }
+
+    this._fetchEvents(this._start, this._end, this._selectedCalendars);
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      <app-header-layout has-scrolling-region>
+        <app-header fixed slot="header">
+          <app-toolbar>
+            <ha-menu-button
+              .hass=${this.hass}
+              .narrow=${this.narrow}
+            ></ha-menu-button>
+            <div main-title>${this.hass.localize("panel.calendar")}</div>
+            <ha-button-menu .icon=${"hass:dots-vertical"}>
+              <mwc-list-item @click=${this._handleRefresh}
+                >Refresh</mwc-list-item
+              >
+            </ha-button-menu>
+          </app-toolbar>
+        </app-header>
+        <div class="content">
+          <div class="calendar-list">
+            <div class="calendar-list-header">My Calendars</div>
+            ${this._calendars.map(
+              (selCal) =>
+                html`<div>
+                  <mwc-formfield .label=${selCal.calendar.name}>
+                    <mwc-checkbox
+                      style=${styleMap({
+                        "--mdc-theme-secondary":
+                          selCal.calendar.backgroundColor,
+                      })}
+                      .value=${selCal.calendar.entity_id}
+                      .checked=${selCal.selected}
+                      @change=${this._handleToggle}
+                    ></mwc-checkbox>
+                  </mwc-formfield>
+                </div>`
+            )}
+          </div>
+          <ha-full-calendar
+            .events=${this._events}
+            .narrow=${this.narrow}
+            @view-changed=${this._handleViewChanged}
+          ></ha-full-calendar>
+        </div>
+      </app-header-layout>
+    `;
+  }
+
+  private get _selectedCalendars(): Calendar[] {
+    return this._calendars
+      .filter((selCal) => selCal.selected)
+      .map((cal) => cal.calendar);
+  }
+
+  private async _fetchEvents(start: Date, end: Date, calendars: Calendar[]) {
+    if (!calendars.length) {
+      return;
+    }
+
+    const events = await fetchCalendarEvents(this.hass, start, end, calendars);
+
+    this._events = [...this._events, ...events];
```

Comment:
The merging should be done by the caller of the function, just return the fetched events.